app/Http/Controllers/Apis/Marketplace/DistributionsApiController.php

```diff
@@ -30,8 +32,66 @@ public function __construct(IDistributionRepository $repository, IResourceServer
         parent::__construct($repository, $resource_server_context);
     }
 
+    #[OA\Get(
+        path: "/api/public/v1/marketplace/distros",
+        description: "Get all marketplace distributions (OpenStack implementations)",
+        summary: 'Get all distributions',
+        operationId: 'getAllDistributions',
+        tags: ['Marketplace Distributions'],
+        parameters: [
+            new OA\Parameter(
+                name: 'filter[]',
+                in: 'query',
+                required: false,
+                description: 'Filter expressions in the format field<op>value. Available fields: name, company. Operators: =@, ==, @@.',
+                style: 'form',
+                explode: true,
+                schema: new OA\Schema(
+                    type: 'array',
+                    items: new OA\Items(type: 'string', example: 'name@@ubuntu')
+                )
+            ),
+            new OA\Parameter(
+                name: 'order',
```

Comment:
list the available order fields check https://github.com/OpenStackweb/summit-api/blob/4de69b2ad7c366126f02ff249128e21e0bb59562/app/Http/Controllers/Apis/Marketplace/AbstractCompanyServiceApiController.php#L77